custom_components/ha_hatch/rest_media_entity.py

```diff
@@ -79,36 +83,58 @@ def _update_local_state(self):
     def _find_track(self, sound_mode=None):
         if sound_mode is None:
             sound_mode = self._attr_sound_mode
+
+        tracks = self._get_tracks()
+
+        return next(
+            (track for track in tracks if track.name == sound_mode),
+            None,
+        )
+
+    def _get_tracks(self) -> list[RestMiniAudioTrack] | list[RestPlusAudioTrack] | list[RIoTAudioTrack]:
         if isinstance(self.rest_device, RestMini):
-            return next(
-                (track for track in REST_MINI_AUDIO_TRACKS if track.name == sound_mode),
-                None,
-            )
-        else:
-            return next(
-                (track for track in REST_PLUS_AUDIO_TRACKS if track.name == sound_mode),
```

Comment:
this appears to remove support for gen1 rest plus. why are you merging the gen1 and gen2 entities? As discussed in the api review, history has shown you will not get users to test a gen2 change against their gen1 devices. Please leave gen2 with their riot prefixes like the other gen2 entities.

Reply:
> this appears to remove support for gen1 rest plus. why are you merging the gen1 and gen2 entities?

This change should still support gen1 rest plus - its captured by the implicit `else` return statement here. I initially merged these two media entities because of the sheer amount of duplicate logic/code between the two of them.

> As discussed in the api review, history has shown you will not get users to test a gen2 change against their gen1 devices.

Yup - that was not mentioned prior to me implementing this change (I did both the lib and this change concurrently).  😅

> Please leave gen2 with their riot prefixes like the other gen2 entities.

Sure - do you mind if I clean up the code in both of these media entities? The way the current code (that is, the code in main branch, not in this PR) is, is a bit funky, IMO; specifically:
1. the code in `media_player.py`:
    ```python
        if isinstance(rest_device, RestIot | RestoreIot):
            return RiotMediaEntity(rest_device)
        elif not isinstance(rest_device, RestoreIot):
            return RestMediaEntity(rest_device, config_turn_on_media)
    ```
   Why have the `elif` statement here as opposed to a generic `else`? The two conditions are a bit confusing, because it says "if restiot or restore iot, create gen2 device, otherwise if not restore iot (aka restiot would possibly be valid, if not already returned), create gen1 device". This is a bit confusing. and IMO should be fixed for correctness (i.e. if restiot or restoreiot, create gen2, else create gen1); something like:
    ```python
        if isinstance(rest_device, RestIot | RestoreIot):
            return RiotMediaEntity(rest_device)
        return RestMediaEntity(rest_device, config_turn_on_media)
    ```
2. the base class in `rest_entity.py` has leakage of `RestPlus` specific implementation for turning on - if we want to keep gen1 and gen2 separate, then that should be moved into the specific media entity to prevent leakage into the base class.
3. can we move the methods (or shared code within) that are identical between the two entities into the base class `RestEntity`, or are you opposed to that entirely? I'm specifically thinking:
   1. The `self._attr_supported_features` features that are applicable across both gen1 and gen2 devices - only gen1 enetity for restplus would need overriding
   2. `set_volume_level()` method 
   3. `media_pause()` method
   4. the common code in `_update_local_state()` method

@dahlb let me know your thoughts please! :)

Reply:
Sorry I missed the rest plus return before you pointed it out, looks like an extra line break within the function confused the GitHub diff so that the return was listed after another function was listed as removed. 

1) yeah that else is ugly and should be updated, thanks

2) we can move the turn on method out of the rest entity and into each of it's sub classes, note though you should check references as rest entity is used for more then the media player, one example being the rest light entity.

3) you could make another parent class for identical methods between rest plus and iot entity, but since rest entity is shared between different kinds of entities, things like supported features don't belong there.